src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -901,6 +901,179 @@ BOOST_AUTO_TEST_CASE(GradientFastLSTMLayerTest)
   BOOST_REQUIRE_LE(CheckGradient(function), 0.2);
 }
 
+/**
+ * Testing the overloaded Forward() of the LSTM layer, for retrieving the Cell
+ * State. Besides output, the overloaded function provides read access to cell
+ * state of the LSTM layer.
+ */
+BOOST_AUTO_TEST_CASE(ReadCellStateParamLSTMLayerTest)
+{
+  const size_t rho = 5, inputSize = 3, outputSize = 2;
+
+  // Provide input of all ones.
+  arma::cube input = arma::ones(inputSize, outputSize, rho);
+
+  arma::mat inputGate, forgetGate, outputGate, hidden;
+  arma::mat outLstm, cellLstm;
+
+  // LSTM layer.
+  LSTM<> lstm(inputSize, outputSize, rho);
+  lstm.Reset();
+  lstm.ResetCell(rho);
+
+  // Initialize the weights to all ones.
+  lstm.Parameters().ones();
+
+  arma::mat inputWeight = arma::ones(outputSize, inputSize);
+  arma::mat outputWeight = arma::ones(outputSize, outputSize);
+  arma::mat bias = arma::ones(outputSize, input.n_cols);
+  arma::mat cellCalc = arma::zeros(outputSize, input.n_cols);
+  arma::mat outCalc = arma::zeros(outputSize, input.n_cols);
+
+  for (size_t seqNum = 0; seqNum < rho; ++seqNum)
+  {
+      // Wrap a matrix around our data to avoid a copy.
+      arma::mat stepData(input.slice(seqNum).memptr(),
+          input.n_rows, input.n_cols, false, true);
+
+      // Apply Forward on LSTM layer.
+      lstm.Forward(std::move(stepData), // Input.
+                   std::move(outLstm),  // Output.
+                   std::move(cellLstm), // Cell State.
+                   false); // Don't write into cell State.
```

Comment:
`State` should be lowercase in the above two comments. Sorry for being too OCD. But it is the way I review code.

Reply:
No issue